src/server/zset_family.cc

```diff
@@ -3042,24 +3148,51 @@ void ZSetFamily::GeoRadiusByMember(CmdArgList args, ConnectionContext* cntx) {
   string_view key = ArgS(args, 0);
   // member to latlong, set shape.xy
   string_view member = ArgS(args, 1);
+  bool member_score_set = false;
```

Comment:
Every Dragonfly command has the command parsing part and the execution part. We never start the execution before the parsing is complete and all the arguments are validated.
it seems you start the transaction execution before the parsing is completed. We should avoid it if possible - this just complicates the function flow.